feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/repository/NftRepositoryImpl.kt

```diff
@@ -31,8 +58,13 @@ class NftRepositoryImpl(
     private val jobOrchestrator: JobOrchestrator,
     private val nftDao: NftDao,
     private val exceptionHandler: HttpExceptionHandler,
+    private val storageSharedRequestsBuilderFactory: StorageSharedRequestsBuilderFactory,
+    private val nftTransfersRegistry: NftTransfersRegistry
 ) : NftRepository {
 
+    private val mutex = Mutex()
+    private val subscriptionBuilders: MutableMap<ChainId, StorageSharedRequestsBuilder> = mutableMapOf()
```

Comment:
I understand the reasoning for adding state to repository but in general we want repositories to stay stateless.

I think we've constrainted ourselves by allowing repository to suscribe to a separate nft. It looks resonable to expose `subscribeNftOwnerAccountIds(nftIds:<String>)` to provide an optimized subscription in a clean way. I think its logically sound since we are trying to hide optimization that is tied to a particular data-source (on-chain subscriptions). In that case we can create subscription builders in a scope of  `subscribeNftOwnerAccountIds` without adding state to repository